timemachine/cpp/src/kernels/k_exchange.cu

```diff
@@ -114,38 +119,63 @@ void __global__ k_attempt_exchange_move_targeted(
     const bool accepted = rand[0] < exp(log_acceptance_prob);
     if (atom_idx == 0 && accepted) {
         num_accepted[0]++;
+        int mol_idx = samples[0];
+        // XOR 1 to flip the flag from 0 to 1 or 1 to 0
+        inner_flags[mol_idx] ^= 1;
     }
 
     // If accepted, move the coords into place
-    while (accepted && atom_idx < N) {
-        dest_coords[atom_idx * 3 + 0] = moved_coords[atom_idx * 3 + 0];
-        dest_coords[atom_idx * 3 + 1] = moved_coords[atom_idx * 3 + 1];
-        dest_coords[atom_idx * 3 + 2] = moved_coords[atom_idx * 3 + 2];
+    // Always move the weights
```

Comment:
Unclear to me what this means - should be "coords"?

Reply:
It means that this kernel will always copy the weights. Either copy the before weights to the after weights (avoids an additional memcpy kernel) or if the move was accepted copy the after weights to be the new before.

Will clarify the code with more comments

Reply:
Done in https://github.com/proteneer/timemachine/pull/1225/commits/0a135e3de4f08ba6b584aa1a26d143baff1cdf66